### ISSUE TITLE
JSON field can not query for "not equals"

### ISSUE THREAD

**Issue by njfix6** [prisma/prisma#7836]
### Bug description

I am trying to query a JSON field for not equals similar to how I see in the docs here: https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#return-all-users-where-name-does-not-equal-eleanor

I am using postgres as the backend. 

I have tried the following:
```
const entries = await this.prismaClient.obj.findMany({
      where: {
        values: {
          path: ["test"],
          not: "test",
        },
    },
});
```
and
```
    const entries = await this.prismaClient.obj.findMany({
      where: {
        values: {
          not: {
            path: ["Email"],
            equals: "breezy@getcorrelated.com",
          },
        },
      },
    });
```
and also a few other variations but none of them seem to give me back any results. I have `previewFeatures = ["orderByRelation", "filterJson"]` so this should be good to go. Please let me know if I'm doing something wrong. 

### How to reproduce

1. Create a table obj with a json field `values`.
2. Create one row with json {"test", "test"}. 
3. Create another row with json {"foo", "bar"}
2. Try to grab the data with 
```
 const entries = await this.prismaClient.obj.findMany({
      where: {
        values: {
          path: ["test"],
          not: "test",
        },
    },
});
```
3. No results will come out

### Expected behavior

I would expect the obj with {"foo", "bar"} to appear when running the query above. 

### Environment & setup

- OS: Mac OS,
- Database: PostgreSQL
- Node.js version: v15.4.0

### Prisma Version

2.20.1

**Comment by pantharshit00** [comment]
I can reproduce this with `2.29.0-dev.5`. We need to define behavior here whether this filter will be limited to the `path` key or not. If so the behavior is correct, otherwise it is not.

**Comment by dpetrick** [comment]
This is not a bug, but a side effect of how SQL databases handle the path extraction and nulls.

This is your test data:
- `{"test", "test"}`
- `{"foo", "bar"}`

Your query searches for `path: ["test"], not: "test"`.
What happens: `{"foo", "bar"}` doesn't have the path `"test"` at all, which results in a path extraction result of `NULL` in the database. You may know that any comparator featuring null on any side (like `NULL != "test"`) is always false, so the row `{"foo", "bar"}` is not returned, because it doesn't match because of the nulls. It's the same really in other filter operations: If you have one row with `NULL` and one with `"value"` in one column, and you check for `not: "value"` you will not get the `NULL` row, instead you need to add a compound filter to check for the `NULL` separately: `OR: [ { col: null }, { col: { not: "value" } } ]`.

Tl;dr: You need to write:
```ts
const entries = await this.prismaClient.obj.findMany({
    where: {
        OR: [
            {
                values: {
                    path: ["test"],
                    not: "test",
                },
            },
            {
                values: {
                    path: ["test"],
                    equals: Prisma.DbNull, // Or similar, see the API changes in the Prisma 3 release to filter for nulls.
                },
            }
        ]
    },
});
```
